Update framework

peppy opened the pull request.

peppy committed:
Update inspection rules in line with source generation change

peppy committed:
Update framework

smoogipoo commented:
Before
```
$ for i in {1..5}; do (touch ../osu.Game/OsuGame.cs && dotnet build) | tail -n 1; done
Time Elapsed 00:00:12.19
Time Elapsed 00:00:14.73
Time Elapsed 00:00:10.89
Time Elapsed 00:00:09.87
Time Elapsed 00:00:11.26
```
After
```
$ for i in {1..5}; do (touch ../osu.Game/OsuGame.cs && dotnet build) | tail -n 1; done
Time Elapsed 00:00:06.28
Time Elapsed 00:00:05.78
Time Elapsed 00:00:05.99
Time Elapsed 00:00:05.49
Time Elapsed 00:00:05.12
```

smoogipoo committed:
Fix incorrect dependency caching

These are tested framework-side to throw exceptions, but the
test is ignored for the source generated version because it quietly
allows this scenario.

The source gen project should eventually have an analyser for these
unsupported cases, or we could consider relaxing this particular case.

peppy commented:
TIL `touch`ing the file is enough to trigger a rebuild

peppy commented:
Just for silly factor comparison, m2 air:

Before:

```
Time Elapsed 00:00:08.12
Time Elapsed 00:00:08.23
Time Elapsed 00:00:08.28
Time Elapsed 00:00:08.50
Time Elapsed 00:00:08.45
```

After:

```
Time Elapsed 00:00:04.97
Time Elapsed 00:00:04.63
Time Elapsed 00:00:04.85
Time Elapsed 00:00:04.70
Time Elapsed 00:00:04.66
```
